client/components/gravatar/index.jsx

```diff
@@ -33,6 +42,16 @@ module.exports = React.createClass( {
 		};
 	},
 
+	componentDidMount() {
```

Comment:
I like the `SERIALIZE` and `DESERIALIZE` idea a lot - it's much neater than doing it in `<Gravatar />`, which relies on Gravatars being displayed in order to remove the extra image from the store, which isn't ideal.

I had a couple of issues with it when you first mentioned it, which is why I didn't use it right away:

Issue 1 (no longer an issue): I wasn't decided yet on how I would fix the Chrome bug ([#](https://github.com/Automattic/wp-calypso/pull/8456#issuecomment-256493017)) where the Gravatar isn't re-fetched at the old URL even after the cache time expires. Now I think I'll solve it in the `uploadGravatar` action creator (so I won't need to know when the local image is removed from the store), freeing me up to use the serialization actions.

Issue 2 (still an issue): removing the image after the expiration date relies on the current time, which would make the `tempImage` reducer impure. 

We could solve this by adding a timestamp to the `SERIALIZE` and `DESERIALIZE` actions.

We could also just write an impure reducer in this instance. The current limitations are mostly to do with tools and testing ([ref](https://github.com/reactjs/redux/issues/1639#issuecomment-212107524)), although we may run into problems later if the purity of reducers is relied upon more heavily in Redux or other tools we use.

I also was thinking we could remove the image when the action `USER_RECEIVE` is dispatched, which would remove the image from the store on each hard refresh (if it's expired). It's not better than the serialization idea, though, so probably going to go with that.